### PR TITLE
[Docs] Convert utilities docs to one page #6447

### PR DESCRIPTION
**Pull request by trancenoid**
The task was to move the various sub heading under "utilities" to one page, instead of multiple pages of each sub-heading.

### CODE REVIEW

**Comment by Crissman** [comment]
Please resolve the conflicts with master.

File: docs/source/reference/util.rst [hunk 1]
```diff
@@ -1,12 +1,177 @@
...
+
+.. autosummary::
+   :toctree: generated/
+   :nosignatures:
+
+   chainer.utils.WalkerAlias
+
+.. _reporter:
+
+Reporter
```
Comment (Crissman) [inline docs/source/reference/util.rst]: This headline can be deleted.

File: docs/source/reference/util.rst [hunk 2]
```diff
@@ -1,12 +1,177 @@
...
+--------------
+
+.. currentmodule:: chainer
+
+Reporter
+~~~~~~~~
```
Comment (Crissman) [inline docs/source/reference/util.rst]:
```suggestion
--------
```

File: docs/source/reference/util.rst [hunk 3]
```diff
@@ -1,12 +1,177 @@
...
+   :toctree: generated/
+   :nosignatures:
+
+   chainer.Reporter
+   chainer.get_current_reporter
+   chainer.report
+   chainer.report_scope
+
+Summary and DictSummary
+~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (Crissman) [inline docs/source/reference/util.rst]:
```suggestion
-------------------------
```

**Comment by Crissman** [comment]
@trancenoid, please update from reviews and resolve conflicts.

**Comment by trancenoid** [comment]
I am having trouble with the syncing the fork, I will try to do it ASAP

**Comment by Crissman** [comment]
LGTM. Jenkins, test this please.

**Comment by Crissman** [comment]
Test failures seem unrelated. Merging.

**Comment by kmaehashi** [comment]
This PR has been backported to v6 in #7152.

**Pull request merged**